specification/dartLangSpec.tex

```diff
@@ -8274,11 +8278,12 @@ \subsection{Constants}
 \item
   An expression of the form \code{$e_1$\,==\,$e_2$} is potentially constant
   if $e_1$ and $e_2$ are both potentially constant expressions.
-  It is further constant if both $e_1$ and $e_2$ are constant and
-  either $e_1$ evaluates to an object that is an instance of
-  \code{int}, \code{double}, \code{String}, \code{bool} or \code{Null},
-  or if $e_2$ evaluates to the null object (\ref{null}).
-  %TODO: Consider adding enum instances here.
+  It is further constant if both $e_1$ and $e_2$ are constant, and
+  either $e_1$ has primitive equality
```

Comment:
We define "has primitive equality" on values, not expressions, so `$e_1$ evaluates to an object that has primitive equality or that is an instance of \code{double},`

Reply:
We do have a notion of classes having primitive equality, but I agree that it's all about instances at this location. Fixed!